metals/src/main/scala/scala/meta/internal/metals/ServerCommands.scala

```diff
@@ -434,3 +434,12 @@ case class DebugUnresolvedAttachRemoteParams(
     port: Int,
     buildTarget: String
 )
+
+case class DebugFullyUnresolvedParams(
+    path: String,
```

Comment:
Why not jsut add nullable/optional fields to DebugUnresolvedMainClassParams  and DebugUnresolvedTestClassParams ?

Reply:
Yea true I'll try doing that.

Reply:
Alright, so I've been playing around with this a bit, and I'm getting pretty convinced that just adding some more `@Nullable` fields here isn't the answer. It adds way more complexity in trying to handle all of the various cases, and I'm finding it way easier to manage to just have another case class for it. Just to give a small example:

### Current
```
case class DebugUnresolvedTestClassParams(
    testClass: String,
    @Nullable buildTarget: String = null
)
```

### Possible new
```
case class DebugUnresolvedTestClassParams(
    @Nullable testClass: String,
	@Nullable path: String,
    @Nullable testType: String,
    @Nullable buildTarget: String = null
)
```
Apart from not loving having every field nullable, you get into all sorts of interesting things that we have to handle server side. What if `testType` is set to `testTarget` and you are provided with a `testClass` that doesn't fit that target. What if for type you pass in `testTarget`, but then you pass in a `path` in one target and the `buildTarget` for another target. There are just so many possiblities that you end up making really strict rules on how they are all handled, but then you lose all that info since it's all contained in a class with all nullable fields. I think having a new class for the discover `DebugFullyUnresolvedParams` for this just makes it way cleaner on the server side. I'm open to maybe trying to figure this out more, but I'm not seeing the benefit of having to change the current functionality to account for the new nullables, and also account for all the crazy cases with the class having every field nullable vs just having a separate class for it. wdyt?

Reply:
It it all becomes much more complex because of the change then sure, I agree it might be beneficial to have a separate class. I just thought it might work better for the existing client implementations, but I haven't dug much into it.